### PR TITLE
Introduce async framework and clean up prefetch related logic

### PR DESCRIPTION
**Pull request by jiangliu**
The main change is to build an async framework for storage/worker by using tokio Notify and current thread runtime.
Also clean up data prefetch related logic, those are left over from legacy code.

### CODE REVIEW

**Review by liubogithub — COMMENTED** [review]
can you please elaborate why mpmc is needed here?

**Comment by jiangliu** [comment]
> ~can you please elaborate why mpmc is needed here?~
> 
> I see that mpsc might be used to serve two async workers, but why do we need two workers if it's async already?

Currently we have enabled the async runtime framework, but most internal implementation is still in sync mode.
So we still need multi-threading here before we have convert the whole stack to async mode.

File: storage/src/device.rs [hunk 1]
```diff
@@ -929,6 +928,13 @@ impl BlobDevice {
         Ok(())
     }
 
+    /// Stop the background blob data prefetch task.
```
[the remark this answers is not in the document]
Comment (jiangliu) [reply storage/src/device.rs]: thanks, fixed

File: docs/images/data_prefetch_architecture.drawio [hunk 2]
```diff
@@ -0,0 +1 @@
...6zOaXKGSucpqdzE8m/rnGCW1Yl5o4m3LRLkK+i6RWuhdmlKD2Tmub1RBdltLtJM0LrPMJkKie4wIkeUjksACJi5Jl9i7cYLRk2mu0xIYNBeygHi3pOGAHhsKD9jE46QSD0AHNYBAC0C0Gh8TF6RhU4Ye+QKgl/uH4VBBqqVyor6VtEA6pi18AIegXhNrLJ5ycCELH0EBodQzC7cmLz7Mfov51S81mITzxNCcr5rgnCQPkznHrcP/I+XIdPieVI3HWqQ1H1eceW9XZWmoDTM/HqFy2u6+iOX35+07ZMFinPuZ0RlBVjXRkYbKK9QGRnSGyrYeeBslTH3o6I6i/y7YNSXApkOwMWof69CkGG4eWoyVZ7emJ0Y3erZ15KdA6e2t3qIffFiSnmouvx8kuxtr1GgEGBUDm8NwDSVACZI7PviF56AO4nedvY0ZyYk36NZIyBFx46dyb3irntNA7I008af/qslq781N12fkRe6HTPkaFbJucKr60zFpq21BsnFHu+DX3fCHfGmQC3DcWZGMVNE9iY+ZsUulmPaqqghGuOYHLm12L/CUUG+atq3QFWd2DPk5qRnFNJca7VQ2EYi+EyjtEQ9ZCbO3MeVQXolVh9LsakfjkLZGSFbfRqjXy9aU43rHCSro2W41K0+9eUb0Kiz0CNGotxQHmlsVlsRZOMMTQQhyFE0UkhdOFoecUIJqZARYkW9DSUA0niGBN+cYy6nII+rOuDPi5RYXcpGUjHSs/KcYmp1s8YqzoaWQm5144XSpCk3WOC33MNGZ0r7AA/VyKZnY4i8SyyZDjKrZlZ3EFl/NW69hy5UgvVQT62qUB+yhqBuIOvyjK63ZqblYuT3NJpW6MzUtVN3zXeWy1uPooYF+xuA3r3alf+4zaGZyVwVkZnJXBWRmclZJtGXyVel+F51Dv8Iuf/8bEOfgpNW5I1Wf4Zd1tK9MfnKDDvzkCZScITKy+vaDxZXhBX9EiUZZ38IQGT2jwhAZPqFVPiBucr2fqCu16VgpDeWYewxA5adJpcHr3GS6mFm1nAvHURHmDiUflfe1NPKbBqvMtRrO0pDoYd5t5VN/pn38u/EBIjo5KlDcJSafXXLhtaMZufBogdT5g0jEgeylFa8XenWJ4mwQk7BWQUH9l13+E6ODPUvSfaACwIUSOpDNbGJ6/Ilsf/gU=</diagram></mxfile>
```
Comment (imeoer) [inline docs/images/data_prefetch_architecture.drawio:1]: Do we need to export drawio format to png, and reference it from somewhere (in doc or code)?

Comment (jiangliu) [reply docs/images/data_prefetch_architecture.drawio:1]: We need more work to improve documentation. There are also coming work to document rafs v6 data format. We could handle it then.

**Review by liubogithub — APPROVED** [review]

**Pull request merged**